app/models/blog.rb

```diff
@@ -8,10 +8,16 @@ class Blog < ApplicationRecord
   validates :title, :content, presence: true
 
   scope :published, -> { where('secret = FALSE') }
-
+  scope :visible_user, lambda { |user|
+    if user.present?
+      where('secret = ? OR user_id = ?', false, user.id)
+    else
+      where(secret: false)
+    end
+  }
   scope :search, lambda { |term|
     keyword = "%#{sanitize_sql(term)}%"
-    where('title LIKE ? OR content LIKE ?', keyword, keyword)
+    where('title LIKE :search_term OR content LIKE :search_term', search_term: keyword)
```

Comment:
```suggestion
    where('title LIKE :keyword OR content LIKE :keyword', keyword:)
```

こう書いてもいいかも？（やや高等テクニック）

Reply:
クエリ内のパラメータ名と変数名を一致させることで省略形が使えるのですね！
ご教示ありがとうございます。修正しました。